kitsune/sumo/parser.py

```diff
@@ -244,16 +245,22 @@ def parse(
 
         @email_utils.safe_translation
         def _parse(locale):
-            return super(WikiParser, self).parse(
-                text,
-                show_toc=show_toc,
-                tags=tags or ALLOWED_TAGS,
-                attributes=attributes or ALLOWED_ATTRIBUTES,
-                styles=styles or ALLOWED_STYLES,
-                nofollow=nofollow,
-                strip_comments=True,
-                **kwargs,
-            )
+            try:
+                return super(WikiParser, self).parse(
+                    text,
+                    show_toc=show_toc,
+                    tags=tags or ALLOWED_TAGS,
+                    attributes=attributes or ALLOWED_ATTRIBUTES,
+                    styles=styles or ALLOWED_STYLES,
+                    nofollow=nofollow,
+                    strip_comments=True,
+                    **kwargs,
+                )
+            except Exception as e:
```

Comment:
let's make this a `TypeError`

Reply:
I wanted to catch all errors, not just the specific one in the bug above, to guard against any future parsing bugs we may encounter.

Reply:
let's catch the one we know about instead of silencing everything.